src/index.js

```diff
@@ -22,32 +51,37 @@ export default function css(options = {}) {
         }
       }
 
-      // Track the order that each stylesheet is imported.
-      if (!order.includes(id)) {
-        order.push(id)
-      }
-
       // Keep track of every stylesheet
       // Check if it changed since last render
+      // NOTE: If we are in transform block, we can assume styles[id] !== code, right?
       if (styles[id] !== code && (styles[id] || code)) {
         styles[id] = code
-        changes++
+        hasChanged = true
       }
 
       return ''
     },
     generateBundle(opts, bundle) {
-      // No stylesheet needed
-      if (!changes || options.output === false) {
-        return
+      const ids = []
+
+      // Determine import order of files
+      for (const file in bundle) {
+        const root = bundle[file].facadeModuleId
+        const modules = getCSSModules(root, this.getModuleInfo)
+        ids.push(...Array.from(modules))
       }
-      changes = 0
 
-      // Combine all stylesheets, respecting import order
+      // If the files are imported in the same order and there are no changes
+      // or options.output is false, there is no work to be done
+      if (arraysEqual(prevIds, ids) && !hasChanged || options.output === false) return
+      prevIds = ids
+
       let css = ''
-      for (let x = 0; x < order.length; x++) {
-        const id = order[x]
-        css += styles[id] || ''
+
+      // Combine all stylesheets, respecting import order
+      for (const index in ids) {
+        let id = ids[index]
```

Comment:
```suggestion
      for (const id of ids) {
```

Reply:
I had intended to write this this way, but `buble` throws this error when I do
![image](https://user-images.githubusercontent.com/13838621/105072224-3b5e1280-5a4b-11eb-93a3-ef35354da6f7.png)

Reply:
I can figure out how to make the change suggested in the error if desired, but I didn't want to go that far if it wasn't desired

Reply:
Ow, no problem then.

How about removing buble and only use node8+ APIs?